crates/turbopack-dev/src/ecmascript/merged/update.rs

```diff
@@ -80,21 +81,31 @@ struct EcmascriptModuleEntry {
 }
 
 impl EcmascriptModuleEntry {
-    fn new(id: &ModuleId, code: ReadRef<Code>, chunk_path: &str) -> Self {
+    async fn new(id: &ModuleId, code: ReadRef<Code>, chunk_path: &str) -> Result<Self> {
         /// serde_qs can't serialize a lone enum when it's [serde::untagged].
         #[derive(Serialize)]
         struct Id<'a> {
             id: &'a ModuleId,
         }
         let id = serde_qs::to_string(&Id { id }).unwrap();
-        EcmascriptModuleEntry {
+
+        let map = (*code).clone().cell().generate_source_map().await?;
```

Comment:
That's pretty weird. Run generate_source_map on the original `Vc<Code>` instead and pass that to this method